File: exercises/csharp/exercise-1/OrcVillage/Generator/OrcMother.cs
using System;
using OrcVillage.Database;

namespace OrcVillage.Generator
{
    public class OrcMother
    {
        //random google for Orc names came up with this https://en.uesp.net/wiki/Lore:Orc_Names
        private static readonly string[] Names =
        {
            "Moghakh", "Atulg", "Azuk", "Bagamul", "Bashag", "Bologra", "Borug",
            "Dur", "Dular", "Duma", "Garothmuk", "Garzonk", "Ghoragdush", "Khagra",
            "Lorzub", "Lugrub", "Olumba", "Orakh"
        };

        private readonly Random rnd = new Random();

        private T GetRandomItem<T>(T[] arr)
        {
            return arr[rnd.Next(arr.Length)];
        }

        public Orc GiveBirth()
        {
            return new Orc
            {
                Id = Guid.NewGuid(),
                Born = DateTime.Now,
                Name = GetRandomItem(Names),
                //All orcs are warriors
                Profession = "Warrior"
            };
        }
    }
}

File: src/fsharp/OrcVillage/OrcMother.fs
module OrcVillage.OrcMother

open System

open OrcVillage.Domain

let private rnd = new Random()

//random google for Orc names came up with this https://en.uesp.net/wiki/Lore:Orc_Names
let private names = [|
            "Moghakh"; "Atulg"; "Azuk"; "Bagamul"; "Bashag"; "Bologra"; "Borug";
            "Dur"; "Dular"; "Duma"; "Garothmuk"; "Garzonk"; "Ghoragdush"; "Khagra";
            "Lorzub"; "Lugrub"; "Olumba"; "Orakh"
            |]


module Array =

    let random (x: 'a []) =
        match x.Length with
        | 0 -> failwith "Cannot select from nonempty array"
        | _ -> Array.get x (rnd.Next x.Length)


let giveBirth(): Orc = {
    Id = Guid.NewGuid()
    Name = names |> Array.random
    // All orcs are warriors
    Profession = "Warrior"
    Born = DateTime.Now
 }





